### PR TITLE
fix cdf5 configure option

### PR DESCRIPTION
**Pull request by wkliao**
This PR fixes configure/cmake that fails to disable CDF5 on 64-bit machines when user explicitly disables CDF5. It also errors out when user explicitly enable CDF5 on 32-bit machines (size_t being of size 4 bytes).

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -888,18 +888,20 @@ AC_CHECK_SIZEOF(unsigned long long)
 # Check whether we want to enable CDF5 support.
 AC_MSG_CHECKING([whether CDF5 support should be enabled])
```
Comment (edhartnett) [inline configure.ac]: Change to "whether CDF5 support should be disabled"

**Comment by wkliao** [comment]
If we kept that line, then ENABLE_CDF5 will be set to either ON or OFF. Unlike autoconf which allows us to set to a customized value, AUTO in this case, meaning user did not explicitly set this option.

I am not aware cmake has a similar command-line option to --help that shows all available options. If you know a way to do so, I am interested to learn.

**Comment by DennisHeimbigner** [comment]
cmake -L
althought it is a bit noisy.

**Comment by wkliao** [comment]
Do people actually run that command to check all available options?
I don't know what to add to achieve the same effect as OPTION, if we want to know whether user explicitly enable or disable an option.

**Comment by DennisHeimbigner** [comment]
I have used it but rarely. I suspect however, that people do peruse
a CMakeLists.txt looking for option() commands.

**Comment by wkliao** [comment]
This commit should achieve the same effect of using OPTION.

**Comment by DennisHeimbigner** [comment]
That seems like a good solution. Wonder why the cmake developers
did not support tristate from the get-go. Oh well.

File: .travis.yml [hunk 2]
```diff
@@ -12,19 +12,19 @@ services:
 env:
     matrix:
 # Ubuntu
-        - DOCKIMG=unidata/nctests:serial   USECMAKE=TRUE USEAC=TRUE DISTCHECK=TRUE USE_CC=gcc   AC_COPTS='CFLAGS=-fsigned-char --disable-netcdf-4 --disable-dap-remote-tests --enable-cdf5' COPTS='-DENABLE_NETCDF_4=OFF -DCMAKE_C_FLAGS=-fsigned-char -DENABLE_DAP_REMOTE_TESTS=OFF -DENABLE_CDF5=TRUE' USECP=FALSE CURHOST=docker-gcc-x64-signed
```
Comment (WardF) [inline .travis.yml]: It doesn't matter so we will accept this, but as a point of reference cmake will support 'ON/OFF' and 'TRUE/FALSE' interchangeably.

**Comment by wkliao** [comment]
I thought it was interchangeable, but when I used -DENABLE_CDF5=FALSE, it does not disable CDF5. Adding a MESSAGE command to print out the value of ENABLE_CDF5 and I got its value ON. Please give it a try. I hope it is not just me.

**Comment by WardF** [comment]
I am seeing test failures on ARM that exist with this merge. I am going to revert this pull request for now as I'm about to go out of town and won't be available to debug this.  We can reevaluate this pull request after I return, and figure out what is going on.

Failures can be seen here:

* http://cdash.unidata.ucar.edu/viewTest.php?onlyfailed&buildid=3790

**Pull request merged**